src/main/java/io/aiven/kafka/connect/opensearch/OpensearchSinkTask.java

```diff
@@ -142,8 +142,19 @@ private void ensureIndexOrDataStreamExists(final String index) {
         if (!indexCache.contains(index)) {
             if (!client.indexOrDataStreamExists(index)) {
                 if (config.dataStreamEnabled()) {
-                    LOGGER.info("Create data stream {}", index);
-                    client.createIndexTemplateAndDataStream(index, config.dataStreamTimestampField());
+                    if (config.dataStreamExistingIndexTemplateName().isPresent()) {
+                        String userProvidedTemplate = config.dataStreamExistingIndexTemplateName().get();
+                        if (!client.dataStreamIndexTemplateExists(userProvidedTemplate)) {
+                            LOGGER.info("Create data stream and template {}", index);
+                            client.createIndexTemplateAndDataStream(userProvidedTemplate,
+                                    config.dataStreamTimestampField());
+                        } else if (client.dataStreamIndexTemplateExists(userProvidedTemplate)) {
```

Comment:
You don't need to check again, the `if (!client.dataStreamIndexTemplateExists(userProvidedTemplate)) {` already means the template exists:
```suggestion
                        } else {
```

Reply:
omg, how did I miss it. Fixed.